requirements.txt

```diff
@@ -18,3 +18,4 @@ fsspec
 setuptools ; python_version >= "3.12"
 packaging
 optree>=0.9.1
+lark
```

Comment:
I'm reluctant to add a new dependency to PyTorch.  I also don't think this will work when someone `pip install`'s PyTorch (which uses setup.py).

Can this be done without an external dependency?

Reply:
Let me clarify my comment here:
It's fine to add any kinds of optional dependencies during CI test runs (which requirments-ci.txt/requirements.txt are about), but adding something as a mandatory dependency requires a more in-depth discussion that should include such considerations as platform availability, interoperability with other packages, transitive dependencies, impact on the import time, etc

Reply:
If we're worried about including Lark for the binary we could use Lark to generate a stand alone parser and check that in.

Reply:
Lark is under 1MB, so hopefully that's fine but yes that would always be an alternative.